### PR TITLE
Some improvements for the build process

### PR DESCRIPTION
**Pull request by skonto**
- Makes some make variables configurable form the env. This is more close to bash practices which is very convenient.
- Use #!/usr/bin/env bash for consistency. Some files used it already like tools/init_paths.sh

### CODE REVIEW

**Comment by skonto** [comment]
@susanxhuynh could you review pls?

**Comment by ArtRand** [comment]
Changes look fine. @susanxhuynh and I will test on mac..

File: Makefile [hunk 1]
```diff
@@ -5,32 +5,33 @@ CLI_DIST_DIR := $(BUILD_DIR)/cli_dist
 DIST_DIR := $(BUILD_DIR)/dist
 GIT_COMMIT := $(shell git rev-parse HEAD)
 
-S3_BUCKET := infinity-artifacts
-S3_PREFIX := autodelete7d
+S3_BUCKET ?= infinity-artifacts
+S3_PREFIX ?= autodelete7d
+SPARK_REPO_URL ?= https://github.com/mesosphere/spark
```
Comment (skonto) [inline Makefile]: This allows to build a test image from any repo.

File: Makefile [hunk 2]
```diff
@@ -5,32 +5,33 @@ CLI_DIST_DIR := $(BUILD_DIR)/cli_dist
...
 	mkdir -p $(DIST_DIR)
 	pushd $(DIST_DIR)
 	wget $(SPARK_DIST_URI)
 	popd
 
-HADOOP_VERSION := $(shell jq ".default_spark_dist.hadoop_version" "$(ROOT_DIR)/manifest.json")
+HADOOP_VERSION ?= $(shell jq ".default_spark_dist.hadoop_version" "$(ROOT_DIR)/manifest.json")
 
-SPARK_DIR := $(ROOT_DIR)/spark
+SPARK_DIR ?= $(ROOT_DIR)/spark
```
Comment (skonto) [inline Makefile]: The good thing about this is that if I set: export SPARK_DIR=$(pwd)/../spark
from my cli then the target below will not be invoked (no cling) which allows me to
build from an existing folder.

**Comment by skonto** [comment]
gentle ping @ArtRand ?

**Comment by skonto** [comment]
@ArtRand @susanxhuynh should we merge it?

**Pull request merged**